client/my-sites/site-settings/site-icon-setting/index.jsx

```diff
@@ -40,11 +51,45 @@ class SiteIconSetting extends Component {
 
 	showModal = () => this.toggleModal( true );
 
-	setSiteIcon = ( media ) => {
-		// [TODO]: Handle setting site icon
-		console.log( media ); // eslint-disable-line no-console
+	editSelectedMedia = ( media ) => {
+		if ( media ) {
+			this.props.onEditSelectedMedia();
+		}
+	};
 
+	setSiteIcon = ( error, blob ) => {
 		this.hideModal();
+		this.props.resetAllImageEditorState();
+
+		if ( error || ! blob ) {
+			return;
+		}
+
+		const { siteId } = this.props;
+		const selectedItem = head( MediaLibrarySelectedStore.getAll( siteId ) );
+		if ( ! selectedItem ) {
+			return;
+		}
+
+		const transientMediaId = uniqueId( 'site-icon' );
+
+		MediaActions.add( siteId, {
+			ID: transientMediaId,
+			fileName: `cropped-${ selectedItem.file }`,
+			fileContents: blob
+		} );
+
+		const checkUploadComplete = () => {
+			const transientMedia = MediaStore.get( siteId, transientMediaId );
+			if ( isItemBeingUploaded( transientMedia ) ) {
+				return;
+			}
+
+			MediaStore.off( 'change', checkUploadComplete );
+			alert( transientMedia.ID );
```

Comment:
>Any reason to not just use a `successNotice` here?

In a subsequent pull request this will be replaced by a call to the settings API, and the successful completion of the settings API request is what would trigger the notice.